### PR TITLE
[pipelining] Rename to stage.py and schedules.py

### PR DESCRIPTION
**Pull request by kwen2501**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #128279
* __->__ #128278
* #128273

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang @d4l3k @ezyang @gchanan

### CODE REVIEW

File: test/allowlist_for_publicAPI.json [hunk 1]
```diff
@@ -2610,23 +2610,6 @@
     "pipe_split",
     "pipeline"
   ],
-  "torch.distributed.pipelining.PipelineSchedule": [
```
Comment (H-Huang) [inline test/allowlist_for_publicAPI.json:2613]: do these not need to be re-added? What was their original purpose

**Review by H-Huang — APPROVED** [review]
LGTM

File: test/allowlist_for_publicAPI.json (hunk 1 above)
Comment (kwen2501) [reply test/allowlist_for_publicAPI.json:2613]: Those were jail breaks :)

**Pull request closed without merging**